client/my-sites/themes/index.web.js

```diff
@@ -32,6 +32,13 @@ export default function( router ) {
 				`/design/:vertical(${ verticals })?/:tier(free|premium)?/filter/:filter/:site_id`,
 				validateFilters, siteSelection, singleSite, makeNavigation, makeLayout
 			);
+			if ( config.isEnabled( 'manage/themes/upload' ) ) {
+				router(
+					'/design/upload/:site_id',
+					siteSelection, upload, makeNavigation, makeLayout
+				);
+				// TODO (seear): make `sites` middleware work for single-tree to allow /design/upload
```

Comment:
I think we need to explicitly use the [`sites`](https://github.com/Automattic/wp-calypso/blob/59b2e4350d6c9fc71348acc1a24536f993c511a5/client/my-sites/controller.js#L309) middleware to get the site selection screen. At least that is what [/menus](https://github.com/Automattic/wp-calypso/blob/59b2e4350d6c9fc71348acc1a24536f993c511a5/client/my-sites/menus/index.js#L16) does.